src/core_plugins/kibana/public/dashboard/index.js

```diff
@@ -275,6 +276,16 @@ app.directive('dashboardApp', function (Notifier, courier, AppState, timefilter,
         $state.panels.push(createPanelState(hit.id, 'visualization', getMaxPanelIndex()));
       };
 
+      if ($route.current.params && $route.current.params[DashboardConsts.ADD_VIS_PARAM]) {
+        $scope.addVis({ id: $route.current.params[DashboardConsts.ADD_VIS_PARAM] });
+        $location.search(DashboardConsts.ADD_TO_DASH_PARAM, null);
```

Comment:
I think replace does the right thing, nothing else is updating the url like I am so it'll just reset on the next url change.